### PR TITLE
Add saving skeleton and deduction of savings from wallet

### PR DESCRIPTION
**Pull request by fatclarence**
Part of #131 

**Allows savings to be deducted from wallet persistently**
Save command allows for deduction of amount of money from user's wallet, referenced from `data\savenus.json` .
Deductions made are persistent in the `Wallet`. 

**Addition of savings.json**
Savings will be kept in a separate json file, which has been created but not yet filled up. 

**TODO**
Will ensure that the command allows for saving persistently in the `savings.json` file.
Complete CurrentSavings and SavingsAccount.

### CODE REVIEW

File: src/main/java/seedu/savenus/logic/commands/SaveCommand.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        // add to the savings account in the model.
+        model.addToSavings(this.savingsAmount);
+
+        // deduct from wallet in model
+        model.deductFromWallet(this.savingsAmount);
```
Comment (Raikonen) [inline src/main/java/seedu/savenus/logic/commands/SaveCommand.java]: Maybe can consider swapping this around, coz if there is an exception thrown from deducting the amount from wallet, that amount shouldn't be added to the savings account

File: src/main/java/seedu/savenus/model/savings/Savings.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        savingsAmount = convert(savings);
+    }
+
+    /**
+     * Returns true if a given string is a valid saving representation.
+     */
+    public static boolean isValidSaving(String saving) {
+        if (saving.matches(VALIDATION_REGEX)) {
+            if (Double.parseDouble(saving) > 5000.00) {
+                return false;
```
Comment (Raikonen) [inline src/main/java/seedu/savenus/model/savings/Savings.java]: Does this mean that users cannot save more than 5000 dollars at once?

Comment (fatclarence) [reply src/main/java/seedu/savenus/model/savings/Savings.java]: Ah didn't notice that I set this limit. Will work on it now.

File: src/main/java/seedu/savenus/logic/commands/SaveCommand.java (hunk 1 above)
Comment (fatclarence) [reply src/main/java/seedu/savenus/logic/commands/SaveCommand.java]: Good idea! Thanks for the thorough review @Raikonen  👍

**Comment by Raikonen** [comment]
LGTM :+1:

**Pull request merged**